geth/account/accounts_test.go

```diff
@@ -103,9 +107,386 @@ func TestVerifyAccountPasswordWithAccountBeforeEIP55(t *testing.T) {
+	password string
+	address  string
+	pubKey   string
+	mnemonic string
+}
+
+// reinitMock is for reassigning a new mock node manager to account manager.
+// Stating the amount of times for mock calls kills the flexibility for
+// development so this is a good workaround to use with EXPECT().AnyTimes()
+func (s *ManagerTestSuite) reinitMock() {
+	s.nodeManager = newMockNodeManager(s.T())
+	s.accManager.nodeManager = s.nodeManager
+}
+
+func (s *ManagerTestSuite) TestCreateAccount() {
+	s.reinitMock()
+
+	// Don't fail on empty password
+	s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil)
+	_, _, _, err := s.accManager.CreateAccount(s.password)
+	s.NoError(err)
+
+	s.nodeManager.EXPECT().AccountKeyStore().Return(nil, testErrKeyStore)
+	_, _, _, err = s.accManager.CreateAccount(s.password)
+	s.Equal(testErrKeyStore, err)
+}
+
+func (s *ManagerTestSuite) TestRecoverAccount() {
+	s.reinitMock()
+
+	s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil)
+	addr, pubKey, err := s.accManager.RecoverAccount(s.password, s.mnemonic)
+	s.NoError(err)
+	s.Equal(s.address, addr)
+	s.Equal(s.pubKey, pubKey)
+
+	s.nodeManager.EXPECT().AccountKeyStore().Return(nil, testErrKeyStore)
+	_, _, err = s.accManager.RecoverAccount(s.password, s.mnemonic)
+	s.Equal(testErrKeyStore, err)
+}
+
+func (s *ManagerTestSuite) TestSelectAccount() {
+	s.reinitMock()
+
+	testCases := []struct {
+		name                  string
+		accountKeyStoreReturn []interface{}
+		whisperServiceReturn  []interface{}
+		address               string
+		password              string
+		expectedError         error
+	}{
+		{
+			"success",
+			[]interface{}{s.keyStore, nil},
+			[]interface{}{s.shh, nil},
+			s.address,
+			s.password,
+			nil,
+		},
+		{
+			"fail_keyStore",
+			[]interface{}{nil, testErrKeyStore},
+			[]interface{}{s.shh, nil},
+			s.address,
+			s.password,
+			testErrKeyStore,
+		},
+		{
+			"fail_whisperService",
+			[]interface{}{s.keyStore, nil},
+			[]interface{}{nil, testErrWhisper},
+			s.address,
+			s.password,
+			testErrWhisper,
+		},
+		{
+			"fail_wrongAddress",
+			[]interface{}{s.keyStore, nil},
+			[]interface{}{s.shh, nil},
+			"wrong-address",
+			s.password,
+			ErrAddressToAccountMappingFailure,
+		},
+		{
+			"fail_wrongPassword",
+			[]interface{}{s.keyStore, nil},
+			[]interface{}{s.shh, nil},
+			s.address,
+			"wrong-password",
+			errors.New("cannot retrieve a valid key for a given account: could not decrypt key with given passphrase"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		s.T().Run(testCase.name, func(t *testing.T) {
+			s.reinitMock()
+			s.nodeManager.EXPECT().AccountKeyStore().Return(testCase.accountKeyStoreReturn...).AnyTimes()
+			s.nodeManager.EXPECT().WhisperService().Return(testCase.whisperServiceReturn...).AnyTimes()
+			err := s.accManager.SelectAccount(testCase.address, testCase.password)
+			if testCase.expectedError != nil {
+				s.Equal(testCase.expectedError, err)
+			} else {
+				s.NoError(err)
```

Comment:
It looks like duplicate check.

Reply:
Ah yes, it looks possible to reduce in this case. Changing.